sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -139,6 +141,8 @@ private[sql] class SQLConf extends Serializable {
    */
   private[spark] def codegenEnabled: Boolean = getConf(CODEGEN_ENABLED, "false").toBoolean
 
+  private[spark] def useSqlSerializer2: Boolean = getConf(USE_SQL_SERIALIZER2, "false").toBoolean
```

Comment:
Instead of using a `Boolean` value, how about use a string type says the `class name` of the Serializer?  We probably will add some other Serializer e.g. the codegen version.